crates/cext/src/circuit.rs

```diff
@@ -0,0 +1,544 @@
+                Qubit(*qubits.wrapping_add(2)),
+                Qubit(*qubits.wrapping_add(3)),
+            ],
+            // There are no standard gates > 4 qubits
+            _ => unreachable!(),
+        };
+        let params: &[Param] = match gate.num_params() {
+            0 => &[],
+            1 => &[(*params.wrapping_add(0)).into()],
+            2 => &[
+                (*params.wrapping_add(0)).into(),
+                (*params.wrapping_add(1)).into(),
+            ],
+            3 => &[
+                (*params.wrapping_add(0)).into(),
+                (*params.wrapping_add(1)).into(),
+                (*params.wrapping_add(2)).into(),
+            ],
+            4 => &[
+                (*params.wrapping_add(0)).into(),
+                (*params.wrapping_add(1)).into(),
+                (*params.wrapping_add(2)).into(),
+                (*params.wrapping_add(3)).into(),
+            ],
+            // There are no standard gates that take > 4 params
+            _ => unreachable!(),
+        };
+        circuit.push_standard_gate(gate, params, qargs);
+    }
+    ExitCode::Success
+}
+
+/// @ingroup QkCircuit
+/// Get the number of qubits for a `QkGate`
+///
+/// @param gate The standard gate to get the number of qubits for
+///
+/// # Example
+///
+///     uint32_t num_qubits = qk_gate_num_qubits(QkGate_CCX);
+///
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub extern "C" fn qk_gate_num_qubits(gate: StandardGate) -> u32 {
+    gate.num_qubits()
+}
+
+/// @ingroup QkCircuit
+/// Get the number of params for a `QkGate`
+///
+/// @param gate The standard gate to get the number of qubits for
+///
+/// # Example
+///
+///     uint32_t num_params = qk_gate_num_params(QkGate_R);
+///
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub extern "C" fn qk_gate_num_params(gate: StandardGate) -> u32 {
+    gate.num_params()
+}
+
+/// @ingroup QkCircuit
+/// Append a measurement to the circuit
+///
+/// @param circuit A pointer to the circuit to add the measurement to
+/// @param qubits The ``uint32_t`` for the qubit to measure
+/// @param clbits The ``uint32_t`` for the clbit to store the measurement outcome in
+///
+/// # Example
+///
+///     QkCircuit *qc = qk_circuit_new(100, 1);
+///     qk_circuit_append_measure(qc, 0, 0);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``circuit`` is not a valid, non-null pointer to a ``QkCircuit``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_circuit_append_measure(
+    circuit: *mut CircuitData,
+    qubit: u32,
+    clbit: u32,
+) -> ExitCode {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let circuit = unsafe { mut_ptr_as_ref(circuit) };
+    circuit.push_packed_operation(
+        PackedOperation::from_standard_instruction(StandardInstruction::Measure),
+        &[],
+        &[Qubit(qubit)],
+        &[Clbit(clbit)],
+    );
+    ExitCode::Success
+}
+
+/// @ingroup QkCircuit
+/// Append a reset to the circuit
```

Comment:
Might be better to specify that we're adding a reset gate?
```suggestion
/// Append a reset gate to the circuit
```

Reply:
A reset is not a gate since it's not a unitary operation.